src/abb_robot_client_RWS2/rws.py

```diff
@@ -0,0 +1,1124 @@
+# Copyright 2022 Wason Technology LLC, Rensselaer Polytechnic Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import struct
+import numpy as np
+import io
+import requests
+import datetime
+import re
+import time
+import websocket
+import threading
+
+from typing import Callable, NamedTuple, Any, List, Union, Optional
+from enum import IntEnum
+
+class ABBException(Exception):
+    """
+    Exception returned from ABB controller
+    """
+    def __init__(self, message, code):
+        super(ABBException, self).__init__(message)
+        self.code=code
+
+class RAPIDExecutionState(NamedTuple):#

```

Comment:
What is the purpose of the #'s at the end of all of these functions declaration headers?